SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -703,17 +769,26 @@ - (nullable NSString *)authToken {
     return self.proxy.protocol.authToken;
 }
 
+- (BOOL)requestRequiresEncryption:(__kindof SDLRPCMessage *)request {
+    if (self.permissionManager.permissions[request.name] != nil) {
+        return self.permissionManager.permissions[request.name].requireEncryption;
+    }
+    return NO;
+}
+
 #pragma mark SDL notification observers
 
 - (void)transportDidConnect {
-    SDLLogD(@"Transport connected");
+    if (![self.lifecycleStateMachine isCurrentState:SDLLifecycleStateReady]) {
```

Comment:
This doesn't appear to be related to this PR. Please remove.

Reply:
this is important because when a secure service is started for RPC type, it will recall all the methods when a normal RPC service is started (including `onProxyOpened` and `transportDidConnect`).  actually that fact is the main reason for creating a new `SDLEncryptionManager` and `SDLEncryptionLifecycleManager` class.  we must wait for `handleProtocolStartServiceACKMessage` after `startSecureServiceWithType` is called and this can only be handled properly if it is called after the initial unsecured service is started.  I was thinking we can have all the encryptions handled in EncryptionManager class RPC, Audio, Video to handle it nicely with the `SDLStreamingMediaManager`, but another option is to add `encrypted` flags to a lot of the methods.  I tried both, in the latter I noticed that it seemed to change more files than necessary

Reply:
The methods that should be changed for sending encrypted / non-encrypted RPCs only needs to be `SDLProtocol` right? It just needs to know if the service that it's being sent on is encrypted and then it can encrypt it right before it sends the data off to the transport.

Having the encryption lifecycle manager handle the setting up of the encryption seems like it would be okay, but I don't see any reason for it to be public, and it would just be a factoring out of the `SDLLifecycleManager`'s setup and teardown process.